Derek furst/track status

DerekFurstPitt opened the pull request:
Implemented property status_history. Its a list containing dictionaries which are status change events.

yuanzhou commented on src/schema/schema_triggers.py:
```diff
@@ -556,6 +556,43 @@ def update_file_descriptions(property_key, normalized_type, user_token, existing
+    status_entry['status'] = status
+    status_entry['changed_by_email'] = email
+    status_entry['change_timestamp'] = current_time
+    new_status_history.append(status_entry)
```
The status history list is not guaranteed to be sorted by the event change time, so we'll need to add `new_status` and `previous_status` as part of each status change event.

yuanzhou replied:
Resolved between @DerekFurstPitt and @shirey 
>So rather than doing the limited implementation now and doing a full list down the road, it just made sense to do it like a list now.

yuanzhou reviewed (changes requested):
@DerekFurstPitt

yuanzhou commented on src/app.py:
```diff
@@ -1363,12 +1364,14 @@ def update_entity(id):
 
         # Handle linkages update via `after_update_trigger` methods
-        if has_direct_ancestor_uuids or has_associated_collection_uuid:
+        if has_direct_ancestor_uuids or has_associated_collection_uuid or has_updated_status:
```
This `has_updated_status` flag and check should also be added inside the `elif normalized_entity_type == 'Upload':` section, otherwise the defined `Upload.status.after_update_trigger.set_status_history` will not execute.

DerekFurstPitt commented:
Updated to accommodate uploads. In addition to adding `or_has_updated_status` under uploads, I would also need to include the logic that sets that property as true, so instead I just moved that block up one level so it will work for datasets/publications and uploads. Performed some tests locally, uploads work and datasets/publications appear to still be working.
